go/inst/instance_topology.go

```diff
@@ -1675,6 +1677,33 @@ func TakeSiblings(instanceKey *InstanceKey) (instance *Instance, takenSiblings i
 	return instance, len(relocatedReplicas), err
 }
 
+// Created this function to allow a hook to be called after a successful TakeMaster event
+func TakeMasterHook(master *Instance, slave *Instance) {
```

Comment:
This repo uses `replica` terminology; can you please change `s/slave/replica`?

Reply:
I've made some changes taking into consideration your other suggestions below, so for consistency, I'll be changing this to successor and demoted (instead of master/slave). Hope this is ok.

Reply:
I've pushed the changes, please let me know if this is ok. Thanks for all the help and input 😄  👍